src/sentry/models/identity.py

```diff
@@ -36,11 +36,43 @@ class IdentityProvider(Model):
     """
 
     __include_in_export__ = False
+    organization = FlexibleForeignKey("sentry.Organization", blank=True, null=True)
 
+    # To be renamed to "provider" because this is a bad column name
     type = models.CharField(max_length=64)
+    # Replacing "type"
+    provider = models.CharField(max_length=64, null=True)
+
+    # To be renamed to "provider_id", because we're already renaming "type"
+    external_id = models.CharField(max_length=64, null=True)
+    # Replacing "external_id"
+    provider_id = models.CharField(max_length=64, null=True)
     config = EncryptedJsonField()
+
     date_added = models.DateTimeField(default=timezone.now, null=True)
-    external_id = models.CharField(max_length=64, null=True)
+
+    # SSO-specific fields
+    is_sso = models.BooleanField(default=False)
+    sso_default_team = models.OneToOneField(
+        "sentry.Team", blank=True, null=True, on_delete=models.SET_NULL
+    )
+    sso_default_role = BoundedPositiveIntegerField(default=0)
+    sso_default_global_access = models.BooleanField(default=False)
```

Comment:
All of this is configuration and should just be part of the data. These shouldn't need to be columns because we're not going to be querying on any of this.

Reply:
I don't really agree with this. If these are important attributes of the provider it makes sense for them to be fields on the model, rather than a schemaless blob

Reply:
The `IdentityProvider` model is designed to generically represent an external provider. Configuration such as auth tokens and other information about how to _use_ the specific provider is stored in the json field. the `type`s of IDPs are represented as provider classes in the `sentry.identity` module (which is where the auth provider stuff should end up moving.

I think the configuration that these attributes represent *for the SSO type of identity provider* does not need to be explicit fields on the model which many identity providers will not even use.

Reply:
Tokens etc make sense, especially since it's an encrypted field. I don't know enough about the specifics of these other config values, but if these attributes will be used across multiple providers I think it's better for them to be columns, even if they're null sometimes.

Reply:
I think there should be distinct separation between provider-related config and Sentry-related config. 

These columns hold SSO defaults, which are relevant to Sentry application only. I've moved into a separate table for now.

The `config` field on this table is used for storing data coming from the providers and the schema varies based on the provider. IMO mixing both types of config data is a bad idea.